kasparovsyncd/sync/transaction_input.go

```diff
@@ -38,47 +28,40 @@ func insertTransactionInputs(dbTx *gorm.DB, transactionIDsToTxsWithMetadata map[
 
 		newNonCoinbaseTransactions[transactionID] = transaction
 		inputsCount += len(transaction.verboseTx.Vin)
-		for _, txIn := range transaction.verboseTx.Vin {
-			outpointsSet[outpoint{
-				transactionID: txIn.TxID,
-				index:         txIn.Vout,
-			}] = struct{}{}
+		for i := range transaction.verboseTx.Vin {
+			txIn := transaction.verboseTx.Vin[i]
+			outpoint := dbaccess.Outpoint{
+				TransactionID: txIn.TxID,
+				Index:         txIn.Vout,
+			}
+			outpointsSet[outpoint] = struct{}{}
 		}
 	}
 
 	if inputsCount == 0 {
 		return nil
 	}
+	outpoints := make([]*dbaccess.Outpoint, 0, len(outpointsSet))
+	for outpoint := range outpointsSet {
+		outpointCopy := outpoint // since outpoint is a value type - copy it, othewise it would be overwritten
+		outpoints = append(outpoints, &outpointCopy)
+	}
 
-	outpoints := outpointSetToSQLTuples(outpointsSet)
-
-	var dbPreviousTransactionsOutputs []*dbmodels.TransactionOutput
-	// fetch previous transaction outputs in chunks to prevent too-large SQL queries
-	for i := 0; i < len(outpoints)/chunkSize+1; i++ {
-		var dbPreviousTransactionsOutputsChunk []*dbmodels.TransactionOutput
-
-		dbResult := dbTx.
-			Joins("LEFT JOIN `transactions` ON `transactions`.`id` = `transaction_outputs`.`transaction_id`").
-			Where("(`transactions`.`transaction_id`, `transaction_outputs`.`index`) IN (?)", outpointsChunk(outpoints, i)).
-			Preload("Transaction").
-			Find(&dbPreviousTransactionsOutputsChunk)
-		dbErrors := dbResult.GetErrors()
-		if httpserverutils.HasDBError(dbErrors) {
-			return httpserverutils.NewErrorFromDBErrors("failed to find previous transaction outputs: ", dbErrors)
-		}
-
-		dbPreviousTransactionsOutputs = append(dbPreviousTransactionsOutputs, dbPreviousTransactionsOutputsChunk...)
+	dbPreviousTransactionsOutputs, err := dbaccess.TransactionOutputsByOutpoints(dbTx, outpoints)
+	if err != nil {
+		return err
 	}
 
 	if len(dbPreviousTransactionsOutputs) != len(outpoints) {
+		log.Infof("len(dbPreviousTransactionsOutputs): %d,  len(outpoints): %d", len(dbPreviousTransactionsOutputs), len(outpoints))
```

Comment:
Why it's Infof? and why it's not part of the error?

Reply:
Because it's a message I added for debugging something and than forgot to delete it :P